packages/package-json-lint/src/rules/require-no-hard-coded-dependency-versions.js

```diff
@@ -0,0 +1,23 @@
+module.exports = {
+  create: ({ ruleConfig, projectType, report }) => ({
+    dependencies: (dependencies) => {
+      if (projectType === 'module' || projectType === 'devModule') {
+        const messageString = 'no hard coded dependency';
+        const currentProblems = Object.keys(dependencies).map(dependencyName => {
+          const dependencyVersion = dependencies[dependencyName];
+          if (!dependencyVersion.startsWith('^') && !(ruleConfig.severity.allowList && ruleConfig.severity.allowList.includes(dependencyName))) {
+            return `${dependencyName}@${dependencyVersion} does not satisfy requirement for ${messageString}`;
```

Comment:
Should we update `messageString` to call out which rule it does not satisfy instead of `no hard coded dependency`?

`${dependencyName}@${dependencyVersion} does not satisfy requirement for the require-no-hard-coded-dependency-versions rule;`

Reply:
We should also add documentation for each rule explaining why we set it up and what problems it helps teams avoid.

Reply:
> Should we update messageString to call out which rule it does not satisfy instead of no hard coded dependency?

Updated here https://github.com/cerner/terra-toolkit/pull/704/commits/b3446c5a790a1172c0c86da818d7d21076835b53.

Reply:
>We should also add documentation for each rule explaining why we set it up and what problems it helps teams avoid.

This is what I was thinking for the documentation.
<img width="1680" alt="Screenshot 2021-09-13 at 3 23 54 PM" src="https://user-images.githubusercontent.com/29273628/133063713-ffed25ec-dfed-4585-ba8f-da4437cea7ca.png">

Reply:
Added documentation here https://github.com/cerner/terra-toolkit/pull/704/commits/0095a1f56d6e9489a7db304cdb5ec3013a9917c9